queries.md

```diff
@@ -3,41 +3,88 @@
 # Answers
 
 ### 1. All the companies that it's name match 'Babelgum'. Retrieve only their `name` field.
+filter : {name: "Babelgum"}
+project : {name: 1, _id:0}
 
 ### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
+filter: {number_of_employees: {$gt:5000}}
+sort: {number_of_employees:1}
+limit: 20
 
 ### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fileds.
+filter: {$and: [{founded_year: {$gt: 1999}}, {founded_year: {$lt: 2006}}]}
+project: {name:1 , founded_year:1, _id:0 }
 
 ### 4. All the companies that had an IPO of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
+filter: {$and: [{ipo: {$gt: 100000000}},{founded_year:{$lt: 2010}}]}
+project: {name:1,ipo:1,_id:0}
 
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
+filter: {$and:[{number_of_employees:{$lt: 1000}}, {founded_year:{$lt:2005}}]}
+sort: {number_of_employees:1}
+limit: 10
 
 ### 6. All the companies that don't include the `partners` field.
+filter: { partners: { $exists: false } }
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
+filter: { category_code: { $type: 'null' } }
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
+filter: {$and: [{number_of_employees:{$gte: 100}}, {number_of_employees:{$lte:1000}}]}
+project:{name:1,number_of_employees:1,_id:0}
 
 ### 9. Order all the companies by their IPO price descendently.
+sort: {'ipo.valuation_amount':-1}
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
+sort: {number_of_employees:-1}
+limit:10
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
+filter: {founded_month: {$gt: 6}}
+limit: 1000
 
-### 12. All the companies that have been 'deadpooled' after the third year.
+### 12. All the companies that have been 'deadpooled' after the third year.1
+
+filter: {$where: 'this.deadpooled_year - this.founded_year > 3'}
 
```

Comment:
La query completa, teniendo en cuenta que algunos de los datos no eran consistentes con el formato año, es la siguiente:
`{$and: [{$where: "(this.deadpooled_year - this.founded_year) > 3"}, {deadpooled_year: {$ne: null}},{founded_year: {$ne: null}}]}`